wdl2cwl/main.py

```diff
@@ -309,99 +311,113 @@ def load_wdl_workflow(self, obj: WDL.Tree.Workflow) -> cwl.Workflow:
-                            else:
-                                input_expr = self.get_step_input_expr(value)  # type: ignore[arg-type]
-                                inputs_from_call[key] = (
-                                    input_expr.replace(".", "/"),
-                                    {},
-                                )
-                wf_step_inputs: List[cwl.WorkflowStepInput] = []
-                for inp in cwl_callee_inputs:
-                    call_inp_id = f"{call.name}.{inp.id}"
-                    source_str, extras = inputs_from_call.get(
-                        cast(str, inp.id), (call_inp_id, {})
-                    )
-
-                    if inp.id not in input_defaults:
-                        wf_step_inputs.append(
-                            cwl.WorkflowStepInput(
-                                id=inp.id, source=source_str, **extras
-                            )
-                        )
-                    else:
-                        wf_step_inputs.append(
-                            cwl.WorkflowStepInput(
-                                id=inp.id, default=source_str, **extras
-                            )
-                        )
-                wf_step_outputs = (
-                    [
-                        cwl.WorkflowStepOutput(id=output.name)
-                        for output in callee.outputs
-                    ]
-                    if callee.outputs
-                    else []
-                )
-                wf_step_run = self.load_wdl_objects(callee)
-                wf_step = cwl.WorkflowStep(
-                    wf_step_inputs,
-                    id=call.name,
-                    run=wf_step_run,
-                    out=wf_step_outputs,
-                )
-                wf_steps.append(wf_step)
+            with WDLSourceLine(body_part, ConversionException):
+                if isinstance(body_part, WDL.Tree.Call):
+                    wf_steps.append(self.get_workflow_call(body_part))
+                elif isinstance(body_part, WDL.Tree.Scatter):
+                    is_scatter_present = True
+                    wf_steps.extend(self.get_workflow_scatter(body_part))
+        requirements = [cwl.ScatterFeatureRequirement()] if is_scatter_present else None
 
         return cwl.Workflow(
             id=wf_name,
             cwlVersion="v1.2",
             doc=wf_description,
+            requirements=requirements,
             inputs=inputs,
             steps=wf_steps,
             outputs=outputs,
         )
 
+    def get_workflow_scatter(self, scatter: WDL.Tree.Scatter) -> List[cwl.WorkflowStep]:
+        """Get the CWL Workflow Step equivalent of a list of WDL Scatter Object."""
+        scatter_steps: List[cwl.WorkflowStep] = []
+        for body_obj in scatter.body:
+            if isinstance(body_obj, WDL.Tree.Call):
+                wp_step = self.get_workflow_call(body_obj)
+            wp_step.scatter = wp_step.in_[0].id
```

Comment:
I'm pretty sure this won't work for any step that is not `scatter`ing the first input.

Reply:
Here is proof

Make the following change to the order of the inputs of the `align_and_count` Task:

``` diff
diff --git a/wdl2cwl/tests/wdl_files/tasks/tasks_reports.wdl b/wdl2cwl/tests/wdl_files/tasks/tasks_reports.wdl
index 6e02e8c..b85b7d2 100644
--- a/wdl2cwl/tests/wdl_files/tasks/tasks_reports.wdl
+++ b/wdl2cwl/tests/wdl_files/tasks/tasks_reports.wdl
@@ -274,8 +274,8 @@ task fastqc {
 
 task align_and_count {
   input {
-    File   reads_bam
     File   ref_db
+    File   reads_bam
     Int    topNHits = 3
 
     Int?   machine_mem_gb
```

`make wdl2cwl/tests/cwl_files/align_and_count_multiple_report.cwl` and check the diff, this changes the `scatter` line:

``` diff
diff --git a/wdl2cwl/tests/cwl_files/align_and_count_multiple_report.cwl b/wdl2cwl/tests/cwl_files/align_and_count_multiple_report.cwl
index 7790762..257fe56 100644
--- a/wdl2cwl/tests/cwl_files/align_and_count_multiple_report.cwl
+++ b/wdl2cwl/tests/cwl_files/align_and_count_multiple_report.cwl
@@ -143,7 +143,7 @@ steps:
         baseCommand:
           - bash
           - script.bash
-    scatter: reads_bam
+    scatter: ref_db
   - id: align_and_count_summary
     in:
       - id: counts_txt
```